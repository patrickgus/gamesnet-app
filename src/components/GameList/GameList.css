.searchBar {
  display: flex;
  align-items: center;
  margin-top: 15px;
  width: 340px;
  color: rgb(0, 224, 224);
}

.searchBar label {
  margin-right: 5px;
}

#search {
  width: 114px;
  border: 1px solid rgb(19, 37, 85);
  border-radius: 4px;
  padding-left: 5px;
  margin-right: 5px;
}

#sort {
  border: 1px solid rgb(19, 37, 85);
  border-radius: 4px;
  width: 60px;
}

@media screen and (max-width: 350px) {
  .searchBar {
    width: auto;
  }

  #search {
    width: 80px;
  }
}

@media screen and (min-width: 680px) {
  .searchBar {
    margin: 15px auto 0;
    width: 480px;
  }

  #search {
    width: 250px;
    margin-right: 10px;
  }

  #sort {
    width: auto;
  }
}

@media screen and (min-width: 1180px) {
  .searchBar label {
    font-size: 20px;
  }
}
