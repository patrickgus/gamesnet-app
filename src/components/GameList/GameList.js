import React, { Component } from "react";
import { Section } from "../Utils/Utils";
import GameListItem from "../../components/GameListItem/GameListItem";
import "./GameList.css";

export default class GameList extends Component {
  constructor(props) {
    super(props);
    this.state = {
      filtered: [],
      sort: ""
    };
  }

  componentDidMount() {
    this.setState({
      filtered: this.props.games
    });
  }

  componentDidUpdate(prevProps) {
    if (this.props.games !== prevProps.games) {
      this.setState({
        filtered: this.props.games
      });
    }
  }

  handleSearch = e => {
    let currentList = [];
    let newList = [];

    if (e.target.value === "") {
      newList = this.props.games;
    } else {
      currentList = this.props.games;

      newList = currentList.filter(game => {
        const title = game.title.toLowerCase();
        const filter = e.target.value.toLowerCase();
        return title.includes(filter);
      });
    }
    this.setState({ filtered: newList });
  };

  handleSort = e => {
    const { filtered } = this.state;
    const sort = e.target.value;

    if (sort === "title") {
      filtered.sort((a, b) => {
        return a[sort] > b[sort] ? 1 : a[sort] < b[sort] ? -1 : 0;
      });
    } else if (sort === "avg_rating") {
      filtered.sort((a, b) => {
        return b[sort] - a[sort];
      });
    }
    this.setState({ sort });
  };

  renderGames() {
    const { filtered } = this.state;
    return filtered.map(game => <GameListItem key={game.id} game={game} />);
  }

  render() {
    const { filtered } = this.state;
    return (
      <Section list className="GameList">
        <div className="searchBar">
          <label htmlFor="search">Search: </label>
          <input
            type="text"
            id="search"
            name="search"
            onChange={this.handleSearch}
          />
          <label htmlFor="sort">Sort: </label>
          <select id="sort" name="sort" onChange={this.handleSort}>
            <option value="title">Title</option>
            <option value="avg_rating">Rating</option>
          </select>
        </div>
        {filtered.length ? (
          this.renderGames()
        ) : (
          <p className="noResults">No matches found. Please try another search.</p>
        )}
      </Section>
    );
  }
}
