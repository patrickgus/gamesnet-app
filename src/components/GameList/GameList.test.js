import React from "react";
import ReactDOM from "react-dom";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import GameList from "./GameList";
import games from "../../stores/game-store";

describe("GameList component", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <BrowserRouter>
        <GameList games={games} />
      </BrowserRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders the form as expected", () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <GameList games={games} />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
