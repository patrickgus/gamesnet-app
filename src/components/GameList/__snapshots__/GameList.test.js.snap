// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameList component renders the form as expected 1`] = `
<section
  className="Section Section--list GameList"
>
  <div
    className="searchBar"
  >
    <label
      htmlFor="search"
    >
      Search: 
    </label>
    <input
      id="search"
      name="search"
      onChange={[Function]}
      type="text"
    />
    <label
      htmlFor="sort"
    >
      Sort: 
    </label>
    <select
      id="sort"
      name="sort"
      onChange={[Function]}
    >
      <option
        value="title"
      >
        Title
      </option>
      <option
        value="avg_rating"
      >
        Rating
      </option>
    </select>
  </div>
  <section
    className="Section GameListItem"
  >
    <header
      className="GameListItem__header"
    >
      <a
        className="GameListItem__link"
        href="/review/1"
        onClick={[Function]}
      >
        <h2
          className="GameListItem__heading"
        >
          First test game
        </h2>
        <img
          alt="Game cover art for First test game"
          className="GameListItem__cover"
          src=""
        />
      </a>
      <h3>
        Average Rating: 
        9
      </h3>
    </header>
    <div
      className="GameListItem__info"
    >
      <h4>
        Description
      </h4>
      <p>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent aliquet, odio vitae imperdiet finibus, purus nisi tincidunt lorem, feugiat bibendum mi mi sit amet urna. Donec a elementum nunc, sed dapibus augue. Duis volutpat at nisl eget varius. Nam vitae libero auctor, lacinia quam ac, fermentum dolor. Donec arcu neque, faucibus eu porta nec, laoreet sit amet eros.
      </p>
      <p>
        <strong>
          Rated:
        </strong>
         
        M
      </p>
      <p>
        <strong>
          Platforms:
        </strong>
         
        Playstaion
      </p>
    </div>
  </section>
  <section
    className="Section GameListItem"
  >
    <header
      className="GameListItem__header"
    >
      <a
        className="GameListItem__link"
        href="/review/2"
        onClick={[Function]}
      >
        <h2
          className="GameListItem__heading"
        >
          Second test game
        </h2>
        <img
          alt="Game cover art for Second test game"
          className="GameListItem__cover"
          src=""
        />
      </a>
      <h3>
        Average Rating: 
        7
      </h3>
    </header>
    <div
      className="GameListItem__info"
    >
      <h4>
        Description
      </h4>
      <p>
        Sed pharetra, ante vel aliquet viverra, augue nunc malesuada dui, vel sollicitudin enim sem at urna. Integer pharetra sodales dolor, id accumsan purus aliquam sed. Donec at magna convallis, varius turpis in, elementum eros. Aliquam erat volutpat. Ut at leo eu ligula fermentum fringilla malesuada ut nisl.
      </p>
      <p>
        <strong>
          Rated:
        </strong>
         
        T
      </p>
      <p>
        <strong>
          Platforms:
        </strong>
         
        Playstation, Xbox, PC, Mobile
      </p>
    </div>
  </section>
  <section
    className="Section GameListItem"
  >
    <header
      className="GameListItem__header"
    >
      <a
        className="GameListItem__link"
        href="/review/3"
        onClick={[Function]}
      >
        <h2
          className="GameListItem__heading"
        >
          Third test game
        </h2>
        <img
          alt="Game cover art for Third test game"
          className="GameListItem__cover"
          src=""
        />
      </a>
      <h3>
        Average Rating: 
        8
      </h3>
    </header>
    <div
      className="GameListItem__info"
    >
      <h4>
        Description
      </h4>
      <p>
        Mauris erat justo, facilisis eget lacus convallis, feugiat luctus massa. Nullam molestie ullamcorper nisl sed posuere. Aliquam erat volutpat. Morbi sed suscipit neque, eu fermentum dolor. Vivamus luctus, eros eu finibus auctor, arcu erat pretium sem, ut tempor mi quam vitae dolor. Mauris feugiat mi in imperdiet commodo.
      </p>
      <p>
        <strong>
          Rated:
        </strong>
         
        M
      </p>
      <p>
        <strong>
          Platforms:
        </strong>
         
        Xbox, Windows
      </p>
    </div>
  </section>
</section>
`;
